persistence/src/main/java/org/opentestsystem/delivery/testreg/service/impl/TestPackageImportServiceImpl.java

```
@@ -30,14 +31,21 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 import tds.common.ValidationError;
+import tds.testpackage.model.TestPackage;
 
 @Service("testPackageImportService")
 public class TestPackageImportServiceImpl implements TestPackageImportService {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(TestPackageImportServiceImpl.class);
 
+    private static Map<String, String> subjectAbbreviationsToLabels = ImmutableMap.of(
+        "ELA", "English Language Arts",
```

Part of me wants to maybe define this as an enum, but I don't know where else it'd be used outside of this class.

---

It is not restricted by a constraint in the XSD, but I agree it could be an enum. As it is now, these mappings exist in the loader perl script that is used with the legacy test package. I think we have a couple of ways to handle this

1. We can update the test package spec to require both a "subject" and "subjectAbbreviation" attribute. Then, we wouldn't have to guess what the mapping is. 
2. We can report an error in the `ValidationStep` in TDS_SupportTool if the subject is unrecognized

I don't think enforcing it via the XSD is a good idea. As it is now, I don't know that SBAC plans on supporting more than just ELA and MATH subjects, but perhaps it would be a good idea to ask.